packages/app/src/dimension-mapper/utils.ts

```diff
@@ -2,10 +2,31 @@ import { type Axis } from '@h5web/shared/vis-models';
 import { useSyncedRef, useUnmountEffect } from '@react-hookz/web';
 import { type DependencyList, useEffect, useMemo, useRef } from 'react';
 
+import { type DimensionMapping } from './models';
+
 export function isAxis(elem: number | Axis): elem is Axis {
   return typeof elem !== 'number';
 }
 
+export function initDimMapping(
+  dims: number[],
+  axesCount: number,
+): DimensionMapping {
+  if (axesCount < 0 || axesCount > 2) {
+    throw new RangeError('Expected 0, 1 or 2 axes');
+  }
+
+  // Cap number of axes to number of dimensions
+  const safeAxesCount = Math.min(axesCount, dims.length);
+
+  return [
+    ...Array.from({ length: dims.length - safeAxesCount }, () => 0),
+    ...(safeAxesCount > 0
```

Comment:
This check used to be `dims.length > 0` and we weren't capping the number of axes to the number of dimensions, so we could end up with the following invalid cases:

- `initDimMapping([], 2)` => `['y', 'x']` even though there are 0 dimensions
- `initDimMapping([1], 2)` => `['y', 'x']` even though there is only 1 dimension
- `initDimMapping([1, 1], 0)` => `['y', 'x']` even though we're specifying that we want a mapping for 0 axes...

Note that I decided to cap the number of axes instead of throwing an error because it's useful for instance in the _Compound_ visualization, which supports both scalar and nD compound datasets. It saves having to compute the right number of axis in the container.